### PR TITLE
feat: allow user to pass keyManager into Txopt

### PR DESCRIPTION
**Pull request by alexgao001**
### Description

add `keyManager` into txOpt, so that allow large batch of txs sent by single client

### Rationale

batch tx sending when switching account

### Example

NA

### Changes

Notable changes: 
* NA

### CODE REVIEW

**Review by Pythonberg1997 — APPROVED** [review]

**Review by guagualvcha — APPROVED** [review]

**Review by guagualvcha — APPROVED** [review]

File: sdk/client/tx.go [hunk 1]
```diff
@@ -101,23 +102,45 @@ func (c *GreenfieldClient) SignTx(ctx context.Context, msgs []sdk.Msg, txOpt *ty
...
+			return nil, err
+		}
+		account, err = c.getAccount()
+		if err != nil {
+			return nil, err
+		}
+		accountNum = account.GetAccountNumber()
+	} else {
+		km = *txOpt.OverrideAccount.Km
+		accountNum = *txOpt.OverrideAccount.Num
```
Comment (guagualvcha) [inline sdk/client/tx.go:124]: `accountNum` seems newly introduced, why don't we only introduce `overide keymanager`, and keep other logic unchanged>

**Review by guagualvcha — CHANGES_REQUESTED** [review]
Need discussion

File: sdk/client/tx.go [hunk 2]
```diff
@@ -101,16 +104,24 @@ func (c *GreenfieldClient) SignTx(ctx context.Context, msgs []sdk.Msg, txOpt *ty
...
+	} else {
+		km, err = c.GetKeyManager()
+		if err != nil {
+			return nil, err
+		}
 	}
-	var nonce uint64
-	account, err := c.getAccount()
+
+	account, err := c.getAccount(&km)
```
Comment (Pythonberg1997) [inline sdk/client/tx.go:121]: I prefer this:
```suggestion
        var nonce uint64
	if txOpt != nil && txOpt.Nonce != 0 {
		nonce = txOpt.Nonce
	} else {
	        account, err := c.getAccount(&km)
	        if err != nil {
		        return nil, err
	        }
	        nonce := account.GetSequence()
	}
```

File: sdk/client/tx.go [hunk 3]
```diff
@@ -242,19 +260,37 @@ func (c *GreenfieldClient) constructTxWithGasInfo(ctx context.Context, msgs []sd
...
+	account, err := c.getAccount(nil)
 	if err != nil {
 		return 0, err
 	}
 	return account.GetSequence(), nil
 }
 
-func (c *GreenfieldClient) getAccount() (authtypes.AccountI, error) {
-	km, err := c.GetKeyManager()
+func (c *GreenfieldClient) GetNonceByKeyManager(km keys.KeyManager) (uint64, error) {
```
Comment (guagualvcha) [inline sdk/client/tx.go:270]: So is this.

**Review by guagualvcha — APPROVED** [review]

**Pull request merged**